### ISSUE TITLE
unbuntu17.04， make runtest error.

### ISSUE THREAD

**Issue by porridgeshoes** [BVLC/caffe#5934]
thanks for carefully read my question.

make all and make test is ok，but when run make runtest is raise an error：

yan@pc:~/download/caffe-master$ make runtest
.build_release/tools/caffe
caffe: command line brew
usage: caffe <command> <args>

commands:
  train           train or finetune a model
  test            score a model
  device_query    show GPU diagnostic information
  time            benchmark model execution time

  Flags from tools/caffe.cpp:
    -gpu (Optional; run in GPU mode on given device IDs separated by ','.Use
      '-gpu all' to run on all available GPUs. The effective training batch
      size is multiplied by the number of devices.) type: string default: ""
    -iterations (The number of iterations to run.) type: int32 default: 50
    -level (Optional; network level.) type: int32 default: 0
    -model (The model definition protocol buffer text file.) type: string
      default: ""
    -phase (Optional; network phase (TRAIN or TEST). Only used for 'time'.)
      type: string default: ""
    -sighup_effect (Optional; action to take when a SIGHUP signal is received:
      snapshot, stop or none.) type: string default: "snapshot"
    -sigint_effect (Optional; action to take when a SIGINT signal is received:
      snapshot, stop or none.) type: string default: "stop"
    -snapshot (Optional; the snapshot solver state to resume training.)
      type: string default: ""
    -solver (The solver definition protocol buffer text file.) type: string
      default: ""
    -stage (Optional; network stages (not to be confused with phase), separated
      by ','.) type: string default: ""
    -weights (Optional; the pretrained weights to initialize finetuning,
      separated by ','. Cannot be set simultaneously with snapshot.)
      type: string default: ""
.build_release/test/test_all.testbin 0 --gtest_shuffle --gtest_filter="-*GPU*"
Note: Google Test filter = -*GPU*
Note: Randomizing tests' orders with a seed of 37730 .
[==========] Running 1106 tests from 150 test cases.
[----------] Global test environment set-up.
[----------] 3 tests from BatchNormLayerTest/1, where TypeParam = caffe::CPUDevice<double>
[ RUN      ] BatchNormLayerTest/1.TestForward
[       OK ] BatchNormLayerTest/1.TestForward (26 ms)
[ RUN      ] BatchNormLayerTest/1.TestForwardInplace
[       OK ] BatchNormLayerTest/1.TestForwardInplace (0 ms)
[ RUN      ] BatchNormLayerTest/1.TestGradient
[       OK ] BatchNormLayerTest/1.TestGradient (185 ms)
[----------] 3 tests from BatchNormLayerTest/1 (211 ms total)

[----------] 1 test from ConstantFillerTest/1, where TypeParam = double
[ RUN      ] ConstantFillerTest/1.TestFill
[       OK ] ConstantFillerTest/1.TestFill (0 ms)
[----------] 1 test from ConstantFillerTest/1 (0 ms total)

[----------] 7 tests from TileLayerTest/1, where TypeParam = caffe::CPUDevice<double>
[ RUN      ] TileLayerTest/1.TestTrivialSetup
[       OK ] TileLayerTest/1.TestTrivialSetup (0 ms)
[ RUN      ] TileLayerTest/1.TestForwardChannels
[       OK ] TileLayerTest/1.TestForwardChannels (0 ms)
[ RUN      ] TileLayerTest/1.TestForwardNum
[       OK ] TileLayerTest/1.TestForwardNum (0 ms)
[ RUN      ] TileLayerTest/1.TestGradientChannels
[       OK ] TileLayerTest/1.TestGradientChannels (253 ms)
[ RUN      ] TileLayerTest/1.TestGradientNum
[       OK ] TileLayerTest/1.TestGradientNum (238 ms)
[ RUN      ] TileLayerTest/1.TestSetup
[       OK ] TileLayerTest/1.TestSetup (1 ms)
[ RUN      ] TileLayerTest/1.TestTrivialGradient
[       OK ] TileLayerTest/1.TestTrivialGradient (77 ms)
[----------] 7 tests from TileLayerTest/1 (569 ms total)

[----------] 1 test from HDF5OutputLayerTest/1, where TypeParam = caffe::CPUDevice<double>
[ RUN      ] HDF5OutputLayerTest/1.TestForward
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.8.17, library is 1.10.0
	    SUMMARY OF THE HDF5 CONFIGURATION
	    =================================

General Information:
-------------------
                   HDF5 Version: 1.10.0-patch1
                  Configured on: Mon Dec  5 18:01:11 UTC 2016
                  Configured by: buildd@lgw01-31
                    Host system: x86_64-pc-linux-gnu
              Uname information: Linux lgw01-31 4.4.0-51-generic #72-Ubuntu SMP Thu Nov 24 18:29:54 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
                       Byte sex: little-endian
             Installation point: /usr
		    Flavor name: serial

Compiling Options:
------------------
                     Build Mode: production
              Debugging Symbols: no
                        Asserts: no
                      Profiling: no
             Optimization Level: high

Linking Options:
----------------
                      Libraries: static, shared
  Statically Linked Executables: 
                        LDFLAGS: -Wl,-Bsymbolic-functions -Wl,-z,relro
                     H5_LDFLAGS: -Wl,--version-script,$(top_srcdir)/debian/map_serial.ver
                     AM_LDFLAGS: 
                Extra libraries: -lpthread -lsz -lz -ldl -lm 
                       Archiver: ar
                         Ranlib: x86_64-linux-gnu-ranlib

Languages:
----------
                              C: yes
                     C Compiler: /usr/bin/gcc
                       CPPFLAGS: -Wdate-time -D_FORTIFY_SOURCE=2
                    H5_CPPFLAGS: -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: 
                        C Flags: -g -O2 -fdebug-prefix-map=/build/hdf5-xxfSIK/hdf5-1.10.0-patch1+docs=. -fstack-protector-strong -Wformat -Werror=format-security
                     H5 C Flags:   -std=c99 -pedantic -Wall -W -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -finline-functions -s -O
                     AM C Flags: 
               Shared C Library: yes
               Static C Library: yes

                        Fortran: yes
               Fortran Compiler: /usr/bin/gfortran
                  Fortran Flags: -g -O2 -fdebug-prefix-map=/build/hdf5-xxfSIK/hdf5-1.10.0-patch1+docs=. -fstack-protector-strong
               H5 Fortran Flags:  -pedantic -Wall -Wextra -Wunderflow -Wimplicit-interface -Wsurprising -Wno-c-binding-type  -s -O2
               AM Fortran Flags: 
         Shared Fortran Library: yes
         Static Fortran Library: yes

                            C++: yes
                   C++ Compiler: /usr/bin/g++
                      C++ Flags: -g -O2 -fdebug-prefix-map=/build/hdf5-xxfSIK/hdf5-1.10.0-patch1+docs=. -fstack-protector-strong -Wformat -Werror=format-security
                   H5 C++ Flags:   -pedantic -Wall -W -Wundef -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wredundant-decls -Winline -Wsign-promo -Woverloaded-virtual -Wold-style-cast -Weffc++ -Wreorder -Wnon-virtual-dtor -Wctor-dtor-privacy -Wabi -finline-functions -s -O
                   AM C++ Flags: 
             Shared C++ Library: yes
             Static C++ Library: yes

                           Java: yes
                  Java Compiler: /usr/bin/java (openjdk 1.8.0_111)

Features:
---------
                  Parallel HDF5: no
             High-level library: yes
                   Threadsafety: yes
            Default API mapping: v18
 With deprecated public symbols: yes
         I/O filters (external): deflate(zlib),szip(encoder)
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
 Packages w/ extra debug output: none
                    API tracing: no
           Using memory checker: no
Memory allocation sanity checks: no
            Metadata trace file: no
         Function stack tracing: no
      Strict file format checks: no
   Optimization instrumentation: no
Bye...
*** Aborted at 1505878566 (unix time) try "date -d @1505878566" if you are using GNU date ***
PC: @     0x7f288f69177f gsignal
*** SIGABRT (@0x3e800000db7) received by PID 3511 (TID 0x7f289291d080) from PID 3511; stack trace: ***
    @     0x7f288fa34670 (unknown)
    @     0x7f288f69177f gsignal
    @     0x7f288f69337a abort
    @     0x7f28917bd320 H5check_version
    @     0x55f557787dd8 caffe::HDF5OutputLayerTest_TestForward_Test<>::TestBody()
    @     0x55f5579dc354 testing::internal::HandleExceptionsInMethodIfSupported<>()
    @     0x55f5579d5b4a testing::Test::Run()
    @     0x55f5579d5c98 testing::TestInfo::Run()
    @     0x55f5579d5d75 testing::TestCase::Run()
    @     0x55f5579d6037 testing::internal::UnitTestImpl::RunAllTests()
    @     0x55f5579d6343 testing::UnitTest::Run()
    @     0x55f5577005e2 main
    @     0x7f288f67c3f1 __libc_start_main
    @     0x55f5577076fa _start
    @                0x0 (unknown)
Aborted (core dumped)
Makefile:532: recipe for target 'runtest' failed
make: *** [runtest] Error 134

**Comment by porridgeshoes** [comment]
i have solve it by reinstall an hdf5
